js/main.js

```diff
@@ -64,13 +68,48 @@ document.addEventListener('click', function (event) {
     }, 2000);
   } else if (event.target.className === 'random-button' || event.target.id === 'random-icon' || event.target.className === 'icon-text random-text') {
     document.querySelector('#home-page').classList.add('hidden');
-
-    document.querySelector('#top-header').textContent = 'Random Cocktail';
+    document.querySelector('#random-ingredients').textContent = '';
     document.querySelector('#random-page').classList.remove('hidden');
     document.querySelector('#browse-page').classList.add('hidden');
     document.querySelector('.home-icon').classList.remove('hidden');
     document.querySelector('.browse').classList.remove('hidden');
     $recipeDiv.innerHTML = '';
+    var idx = random(0, allCocktails.drinks.length - 1);
+    document.querySelector('#random-name').innerHTML = '';
+    document.querySelector('#top-header').textContent = 'Random Cocktail # ' + [idx + 1] + '-' + allCocktails.drinks[idx].strDrink;
+    document.querySelector('#random-img-div').innerHTML = '';
+    document.querySelector('#random-list').textContent = '';
```

Comment:
You don't want to call querySelector every time to query the element.

Elements that you will be reusing you want to store in variables in the global scope.

This is a bad practice and slows down your code.